frontend/src/metabase/static-viz/components/XYChart/utils/series.ts

```diff
@@ -50,16 +50,38 @@ export const sortSeries = (series: Series[], type: XAxisType) => {
   });
 };
 
-export const calculateStackedItems = (series: Series[]) => {
+export const calculateStackedItems = (multipleSeries: Series[]) => {
+  const dimensionSeriesIndexMap = multipleSeries.reduce(
+    (map, series, seriesIndex) => {
+      series.data.forEach(datum => {
+        const dimension = getX(datum);
+        if (!map[dimension]) {
+          map[dimension] = {};
+        }
+        map[dimension][seriesIndex] = datum;
+      });
+      return map;
+    },
+    {} as {
+      [dimension: string]: {
+        [seriesIndex: number]: SeriesDatum;
+      };
+    },
+  );
+
   // Stacked charts work only for a single dataset with one dimension
-  return series.map((s, seriesIndex) => {
-    const stackedData = s.data.map((datum, datumIndex) => {
+  return multipleSeries.map((series, seriesIndex) => {
+    const stackedData = series.data.map((datum, datumIndex) => {
       const [x, y] = datum;
 
       let y1 = 0;
 
       for (let i = 0; i < seriesIndex; i++) {
-        const currentY = getY(series[i].data[datumIndex]);
+        const currentDatum = dimensionSeriesIndexMap[x][i];
+        if (!currentDatum) {
```

Comment:
Should it be `currentDatum != null` instead to not omit zeroes?

Reply:
Thank you! That's right. I'll fix that